### PR TITLE
Add rz-bindgen blog post

### PR DESCRIPTION
**Pull request by wingdeans**
For GSoC 2022

### CODE REVIEW

**Review by XVilka — APPROVED** [review]
Thanks! Looks good as is. Lets wait for others though to take a look.

File: content/posts/gsoc-2022-rz-bindgen/index.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# Implementation
+I considered my primary options for parsing the C headers to be [tree-sitter](https://github.com/tree-sitter/tree-sitter/) and [libclang](https://clang.llvm.org/docs/Tooling.html#libclang). Even though I wrote about tree-sitter in the Rizin GSoC announcement blogpost, the integrated preprocessor and semantic analysis led me to choose [libclang's Python bindings](https://libclang.readthedocs.io/en/latest/).
+
+## C Structs and Functions
+Once a header is parsed, C data structures are grouped with functions that operate on them. In [this snippet from rz-bindgen](https://github.com/rizinorg/rz-bindgen/blob/e010ce8d688cfbb12e99dc15868d818aeda21b5b/src/bindings.py), the `RzAnalysis` struct from the `rz_analysis.h` header is grouped with functions that have the `rz_analysis_` prefix. In the generated Python bindings, these groupings are mapped to object-oriented classes, with the `RzAnalysis` class containing the grouped functions as its methods. The `RzAnalysis` class also makes all the fields of the C struct accessible except for `leaddrs` (which is ignored as per the `ignore_fields` argument) and `type_links` (which is renamed as per the `rename_fields` argument).
```
Comment (thestr4ng3r) [inline content/posts/gsoc-2022-rz-bindgen/index.md:28]:
```suggestion
Once a header is parsed, C data structures are grouped with functions that operate on them. In [this snippet from rz-bindgen](https://github.com/rizinorg/rz-bindgen/blob/e010ce8d688cfbb12e99dc15868d818aeda21b5b/src/bindings.py#L146-L170), the `RzAnalysis` struct from the `rz_analysis.h` header is grouped with functions that have the `rz_analysis_` prefix. In the generated Python bindings, these groupings are mapped to object-oriented classes, with the `RzAnalysis` class containing the grouped functions as its methods. The `RzAnalysis` class also makes all the fields of the C struct accessible except for `leaddrs` (which is ignored as per the `ignore_fields` argument) and `type_links` (which is renamed as per the `rename_fields` argument).
```

File: content/posts/gsoc-2022-rz-bindgen/index.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+rz_analysis.add_method("rz_analysis_reflines_get", rename="get_reflines")
+rz_analysis.add_prefixed_methods("rz_analysis_")
+rz_analysis.add_prefixed_funcs("rz_analysis_")
+```
+
+## Generation
+Rz-bindgen is designed to support multiple backends to generate bindings for a variety of languages. A backend takes the `Class` objects created in the transformation step and generates output. There are, at the time of writing, a [SWIG](https://www.swig.org/) backend and a [Sphinx](https://www.sphinx-doc.org/) backend.
+
+The SWIG backend is currently only used for Python bindings, but SWIG targets other languages, such as Java and OCaml. Supporting them in rz-bindgen should be relatively simple.
```
Comment (thestr4ng3r) [inline content/posts/gsoc-2022-rz-bindgen/index.md:45]:
```suggestion
The SWIG backend is currently only used for Python bindings, but SWIG targets other languages too, such as Java and OCaml. Supporting them in rz-bindgen should be relatively simple.
```

**Review by thestr4ng3r — APPROVED** [review]
Some minor suggestions, overall an excellent post.

**Review by wargio — APPROVED** [review]

File: content/posts/gsoc-2022-rz-bindgen/index.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+def print_function_info(fn: rizin.RzAnalysisFunction):
+    print("name:", fn.name)
+    print("number of xrefs from:", len(fn.get_xrefs_from()))
+    print("number of xrefs to:", len(fn.get_xrefs_to()))
+    return True
+core.register_group("u", "A custom group for user-defined commands")
+core.register_command("uf", print_function_info)
+```
+
+The [Rizin plugin](https://github.com/rizinorg/rz-bindgen/blob/9ccbc56cefaf043b95666793dd1bc156480bbe6c/plugin) registers Python as an `RzLang`, allowing users to load Python scripts on the fly. It also adds add a `core` variable to the `rizin` Python module, allowing scripts that import it to access Rizin's own `RzCore`.
```
Comment (ret2libc) [inline content/posts/gsoc-2022-rz-bindgen/index.md:80]: "Adds add"

Comment (wargio) [inline content/posts/gsoc-2022-rz-bindgen/index.md:80]:
```suggestion
The [Rizin plugin](https://github.com/rizinorg/rz-bindgen/blob/9ccbc56cefaf043b95666793dd1bc156480bbe6c/plugin) registers Python as an `RzLang`, allowing users to load Python scripts on the fly. It also adds a `core` variable to the `rizin` Python module, allowing scripts that import it to access Rizin's own `RzCore`.
```

**Pull request merged**